source/extensions/transport_sockets/proxy_protocol/proxy_protocol.cc

```diff
@@ -165,6 +176,69 @@ void UpstreamProxyProtocolSocketFactory::hashKey(
   }
 }
 
+std::vector<Envoy::Network::ProxyProtocolTLV> UpstreamProxyProtocolSocket::processCustomTLVs() {
+  std::vector<Envoy::Network::ProxyProtocolTLV> custom_tlvs;
+  absl::flat_hash_set<uint8_t> processed_tlv_types;
+
+  processHostLevelTLVs(custom_tlvs, processed_tlv_types);
```

Comment:
I think this will be easier to read if you combine these three functions into one, and then you don't need the in/out parameters.

Reply:
IMO the single function was brutal to read due to the level of nesting needed to make sure that failure to unpack the TLVs from host metadata does not impact the ability to extract TLVs from the transport socket config. Happy to refactor into a single function in the next set of commits though.

Reply:
Refactored into a single function in https://github.com/envoyproxy/envoy/pull/37591/commits/ad46d237feaee356618a904f327601d8f5efac1d.